src/server/sql/reading/create_reading_views.sql

```diff
@@ -209,84 +210,87 @@ CREATE OR REPLACE FUNCTION meter_line_readings_unit (
-			-- it then uses the unit_index for this unit.
-			INNER JOIN meters m ON m.id = meters.id)
-			INNER JOIN units u ON m.unit_id = u.id)
-			-- This is getting the conversion for the meter (row_index) and unit to graph (column_index).
-			-- The slope and intercept are used above the transform the reading to the desired unit.
-			INNER JOIN cik c on c.row_index = u.unit_index AND c.column_index = unit_column)
-			WHERE requested_range @> time_interval;
-	-- There's no quick way to get the number of hours in an interval. extract(HOURS FROM '1 day, 3 hours') gives 3.
-	ELSIF extract(EPOCH FROM requested_interval)/3600 >= min_hour_points THEN
-		-- Get hourly points to graph. See daily for more comments.
-		RETURN QUERY
-			SELECT hourly.meter_id AS meter_id,
-				-- Convert the reading based on the conversion found below.
-				-- Hourly readings are already averaged correctly into a rate.
-				hourly.reading_rate * c.slope + c.intercept as reading_rate,
-				lower(hourly.time_interval) AS start_timestamp,
-				upper(hourly.time_interval) AS end_timestamp
-			FROM ((((hourly_readings_unit hourly
-			INNER JOIN unnest(meter_ids) meters(id) ON hourly.meter_id = meters.id)
-			INNER JOIN meters m ON m.id = meters.id)
-			INNER JOIN units u ON m.unit_id = u.id)
-			INNER JOIN cik c on c.row_index = u.unit_index AND c.column_index = unit_column)
-			WHERE requested_range @> time_interval;
-	 ELSE
-		-- Default to raw/meter data to graph. See daily for more comments.
-		RETURN QUERY
-			SELECT r.meter_id as meter_id,
-			CASE WHEN u.unit_represent = 'quantity'::unit_represent_type THEN
-				-- If it is quantity readings then need to convert to rate per hour by dividing by the time length where
-				-- the 3600 is needed since EPOCH is in seconds.
-				((r.reading / (extract(EPOCH FROM (r.end_timestamp - r.start_timestamp)) / 3600)) * c.slope + c.intercept) 
-			WHEN (u.unit_represent = 'flow'::unit_represent_type OR u.unit_represent = 'raw'::unit_represent_type) THEN
-				-- If it is flow or raw readings then it is already a rate so just convert it but also need to normalize
-				-- to per hour.
-				((r.reading * 3600 / u.sec_in_rate) * c.slope + c.intercept)
-			END AS reading_rate,
-			r.start_timestamp,
-			r.end_timestamp
-			FROM ((((readings r
-			INNER JOIN unnest(meter_ids) meters(id) ON r.meter_id = meters.id)
-			INNER JOIN meters m ON m.id = meters.id)
-			INNER JOIN units u ON m.unit_id = u.id)
-			INNER JOIN cik c on c.row_index = u.unit_index AND c.column_index = unit_column)
-			WHERE lower(requested_range) <= r.start_timestamp AND r.end_timestamp <= upper(requested_range);
-	 END IF;
+	-- Start with the raw, then hourly and then daily if others will not work
+
+	while add <= cardinality(meter_ids) loop
+		select meter_reading_frequency into frequency from meters where id = meter_ids[add];
+		IF (((EXTRACT (DAY FROM requested_interval) * 1440) / (EXTRACT (HOUR FROM frequency) * 60 + EXTRACT (MINUTE FROM frequency))) <= 1440) 
```

Comment:
I cannot convince myself that the logic precisely does what is wanted but is close. It seems to assume that the requesteed_interval is an even number of days and I don't think that is the case. I think this since I thought EXTRTACT DAY only returns whole days. I also think we could have an issue with frequency if someone entered it in the general Postgres syntax of '4 days' rather than '96:00:00' since it does not deal with days. A quick test found this did not work with the current code.

To do this in a general way is probably desirable given that there are variety of ways to enter these values into Postgres. I'm hoping there is a way to do this but I have not investigated.